# Scoring SUS
SUS yields a single number representing a composite measure of the overall usability of the
system being studied. Note that scores for individual items are not meaningful on their own.
To calculate the SUS score, first sum the score contributions from each item. Each item's
score contribution will range from 0 to 4. For items 1,3,5,7,and 9 the score contribution is the
scale position minus 1. For items 2,4,6,8 and 10, the contribution is 5 minus the scale
position. Multiply the sum of the scores by 2.5 to obtain the overall value of SU.
SUS scores have a range of 0 to 100. 

In [19]:
import pandas as pd
usability_data = pd.read_csv('Usabilitysurvey.csv')
usability_data.columns

Index(['id', 'title', 'user_id', 'created', 'usage_frequency',
       'system_complexity', 'usage_ease', 'need_help', 'functions_integrated',
       'system_inconsistency', 'learn_to_use', 'system_inconvenient',
       'confident_level', 'learning_before', 'comment'],
      dtype='object')

In [21]:
for i in usability_data['comment']:
    print(i)

nan
good contribution
nan
nan
I understand you are testing the Q&A system on me. Fine.
I thought there were too many questions after the three lists and almost gave up because i got bored. The questions are too similar.
nan
I know its very easy to suggest countries from the same climat zone or from the same region but try not to.
nan
Well you might add a final list with ratings. But that wouls be considered in a later phase. Good job Ayoub
nan
nan
nan
nan
nan
nan
Good works
nan
The suggested list were pretty much all similar. I deliberatly chose many asian countries but only Indonisia got proposed to me. 

South Korea is not on the list
nan
nan
nan
nan
nan
I feel like the list of countries I got was completely random and not at all in any ways related to the questions I've ansered...
nan
nan
List of countries is not full (no Poland or 2 Koreas);
In mobile view 3rd list is very inconvenient to get to
probably the registration makes it less attractive for people to make this test
nan
May

In [3]:
usability_data = pd.read_csv('Usabilitysurvey.csv')
usability_data = usability_data.loc[:,'usage_frequency':'learning_before']
usability_data_av = usability_data
len(usability_data)

109

In [4]:
usability_data.head()

usage_frequency system_complexity         usage_ease          need_help  \
0  Strongly_Disagree    Strongly_Agree  Strongly_Disagree           Disagree   
1           Disagree             Agree            Neutral           Disagree   
2            Neutral          Disagree              Agree           Disagree   
3            Neutral             Agree            Neutral           Disagree   
4           Disagree             Agree     Strongly_Agree  Strongly_Disagree   

  functions_integrated system_inconsistency learn_to_use system_inconvenient  \
0    Strongly_Disagree                Agree     Disagree      Strongly_Agree   
1                Agree    Strongly_Disagree      Neutral             Neutral   
2              Neutral             Disagree        Agree             Neutral   
3              Neutral              Neutral        Agree            Disagree   
4             Disagree                Agree        Agree             Neutral   

  confident_level    learning_before  
0         Neutral           Disagree  
1         Neutral  Strongly_Disagree  
2         Neutral           Disagree  
3           Agree           Disagree  
4           Agree  Strongly_Disagree

In [5]:
usability_data = usability_data.replace(['Strongly_Agree'], '5')
usability_data = usability_data.replace(['Agree'], '4')
usability_data = usability_data.replace(['Neutral'], '3')
usability_data = usability_data.replace(['Disagree'], '2')
usability_data = usability_data.replace(['Strongly_Disagree'], '1')

In [6]:
usability_data.columns = ['1','2','3','4','5',
                         '6','7','8','9','10']
Data_trans = usability_data.T

In [7]:
## Positively formulated question [1,3,5,7,9] - 1
Data_trans.to_csv('usability_survey.csv')

In [8]:
Data_trans = pd.read_csv('usability_survey.csv')

In [9]:
Data_trans.head(10)

Unnamed: 0  0  1  2  3  4  5  6  7  8  ...  99  100  101  102  103  104  \
0           1  1  2  3  3  2  3  3  3  3  ...   1    3    4    2    4    3   
1           2  5  4  2  4  4  4  1  2  1  ...   3    4    2    2    3    1   
2           3  1  3  4  3  5  4  5  5  3  ...   3    2    4    4    4    4   
3           4  2  2  2  2  1  4  1  1  2  ...   1    2    2    2    2    1   
4           5  1  4  3  3  2  4  5  4  3  ...   3    3    4    4    4    4   
5           6  4  1  2  3  4  2  5  2  2  ...   4    4    2    3    4    4   
6           7  2  3  4  4  4  3  5  4  5  ...   3    3    4    4    5    4   
7           8  5  3  3  2  3  2  2  2  2  ...   3    3    2    2    3    1   
8           9  3  3  3  4  4  3  5  3  2  ...   2    3    4    4    4    4   
9          10  2  1  2  2  1  2  2  2  2  ...   1    4    2    1    2    1   

   105  106  107  108  
0    3    3    3    1  
1    3    2    3    4  
2    3    4    4    2  
3    3    2    2    3  
4    3    4    3    1  
5    3    3    2    4  
6    3    3    4    2  
7    3    2    2    5  
8    3    4    3    3  
9    3    2    2    3  

[10 rows x 110 columns]

In [10]:
positive_question = Data_trans.loc[[0,2,4,6,8],:]
negative_question = Data_trans.loc[[1,3,5,7,9],:]

In [11]:
positive_question = positive_question - 1  # 1 - score
negative_question = 5 - negative_question # 5- score

In [12]:
# compute the score of positive question for each user
score_pos = []
for i in positive_question.columns:
    sum_ = 0
    sum_ = sum(positive_question[i])
    score_pos.append(sum_*2.5)

In [13]:
# compute the score of negative question for each user
score_neg = []
for i in negative_question.columns:
    sum_ = 0
    sum_ = sum(negative_question[i])
    score_neg.append(sum_*2.5)

In [14]:
score_total = []
for i, j in zip(score_neg, score_pos):
    score_total.append(i+j)

In [18]:
score_total
import statistics

print(min(score_total))
print(max(score_total))
print(statistics.mean(score_total))
print(statistics.median(score_total))


22.5
100.0
60.59090909090909
62.5


In [126]:
total_Score = sum(score_total)
sus_score = total_Score / 109
sus_score

60.80275229357798

## The system SUS score = 60.80

## the average of each question

In [119]:
usability_data_av.head()

usage_frequency system_complexity         usage_ease          need_help  \
0  Strongly_Disagree    Strongly_Agree  Strongly_Disagree           Disagree   
1           Disagree             Agree            Neutral           Disagree   
2            Neutral          Disagree              Agree           Disagree   
3            Neutral             Agree            Neutral           Disagree   
4           Disagree             Agree     Strongly_Agree  Strongly_Disagree   

  functions_integrated system_inconsistency learn_to_use system_inconvenient  \
0    Strongly_Disagree                Agree     Disagree      Strongly_Agree   
1                Agree    Strongly_Disagree      Neutral             Neutral   
2              Neutral             Disagree        Agree             Neutral   
3              Neutral              Neutral        Agree            Disagree   
4             Disagree                Agree        Agree             Neutral   

  confident_level    learning_before  
0         Neutral           Disagree  
1         Neutral  Strongly_Disagree  
2         Neutral           Disagree  
3           Agree           Disagree  
4           Agree  Strongly_Disagree

In [120]:
d = {}
for i in usability_data_av.columns:
    d[i] = usability_data_av[i].value_counts()

In [121]:
aver_res = pd.DataFrame.from_dict(d)
aver_res.head()

usage_frequency  system_complexity  usage_ease  need_help  \
Agree                           18                 18          52        6.0   
Disagree                        28                 44          12       39.0   
Neutral                         39                 28          22        9.0   
Strongly_Agree                   1                  7          17        NaN   
Strongly_Disagree               23                 12           6       55.0   

                   functions_integrated  system_inconsistency  learn_to_use  \
Agree                                36                    24            53   
Disagree                             13                    33             8   
Neutral                              47                    37            26   
Strongly_Agree                        6                     9            21   
Strongly_Disagree                     7                     6             1   

                   system_inconvenient  confident_level  learning_before  
Agree                               19               35                8  
Disagree                            43               15               45  
Neutral                             28               43               16  
Strongly_Agree                       7               11                2  
Strongly_Disagree                   12                5               38

In [124]:
for c in aver_res.columns:
    aver_res[c] = aver_res[c] / aver_res[c].sum()
    aver_res[c] = aver_res[c].mul(100).round(1).astype(str) +'%'
#     aver_res = aver_res['usage_frequency'] / aver_res['usage_frequency'].sum()
# aver_res * 100

In [125]:
aver_res.head()

usage_frequency system_complexity usage_ease need_help  \
Agree                       16.5%             16.5%      47.7%      5.5%   
Disagree                    25.7%             40.4%      11.0%     35.8%   
Neutral                     35.8%             25.7%      20.2%      8.3%   
Strongly_Agree               0.9%              6.4%      15.6%      nan%   
Strongly_Disagree           21.1%             11.0%       5.5%     50.5%   

                  functions_integrated system_inconsistency learn_to_use  \
Agree                            33.0%                22.0%        48.6%   
Disagree                         11.9%                30.3%         7.3%   
Neutral                          43.1%                33.9%        23.9%   
Strongly_Agree                    5.5%                 8.3%        19.3%   
Strongly_Disagree                 6.4%                 5.5%         0.9%   

                  system_inconvenient confident_level learning_before  
Agree                           17.4%           32.1%            7.3%  
Disagree                        39.4%           13.8%           41.3%  
Neutral                         25.7%           39.4%           14.7%  
Strongly_Agree                   6.4%           10.1%            1.8%  
Strongly_Disagree               11.0%            4.6%           34.9%